hibernate-core/src/main/java/org/hibernate/boot/model/naming/Identifier.java

```diff
@@ -23,6 +23,7 @@
 public class Identifier implements Comparable<Identifier> {
 	private final String text;
 	private final boolean isQuoted;
+	private boolean isExplicit;
```

Comment:
It would probably be better to keep `Identifier` is a completely immutable object, and overload `Database.toIdentifier()` to accept `isExplicit` is a parameter.